spec/forms/enterprise_fees_bulk_update_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe EnterpriseFeesBulkUpdate do
+  describe "error reporting" do
+    let(:enterprise_fee) { build_stubbed(:enterprise_fee) }
+    let(:base_attributes) do
+      attributes = enterprise_fee.attributes.symbolize_keys
+      attributes[:calculator_type] = enterprise_fee.calculator_type
+      attributes[:calculator_attributes] = enterprise_fee.calculator.attributes
+      attributes
+    end
+    let(:valid_attributes) do
+      set_attributes = {
+        sets_enterprise_fee_set: {
+          collection_attributes: {
+            "0" => base_attributes
+          }
+        }
```

Comment:
This level of specification seems unnecessarily deep. Perhaps next we will extract this functionality out to a form module and test it there.